functorch/csrc/DynamicLayer.h

```diff
@@ -11,16 +11,18 @@ namespace at {
 namespace functorch {
 
 struct TORCH_API DynamicLayer {
-  DynamicLayer(DispatchKey key, int64_t layerId): key_(key), layerId_(layerId) {}
+  DynamicLayer(DispatchKey key, int64_t layerId, optional<int64_t> batchSize = nullopt): key_(key), layerId_(layerId), batchSize_(batchSize) {}
 
   DispatchKey key() const { return key_; }
   int64_t layerId() const { return layerId_; }
+  int64_t batchSize() const { return *batchSize_; }
  private:
   DispatchKey key_;
   int64_t layerId_;
+  optional<int64_t> batchSize_;
```

Comment:
This LGTM to me for now. In the long-term future, the DynamicLayer struct probably shouldn't hold a batchSize_ (the DynamicLayer system shouldn't need to know about implementation details of each transform). Instead we want some way for transform writers to come along and "register" some "per-level metadata".

Reply:
yeah, agreed.